### PR TITLE
update tab navigator and icons and add groups

### PR DESCRIPTION
**Pull request by dereckquock**

### CODE REVIEW

File: navigation/MainTabNavigator.js [hunk 1]
```diff
@@ -141,16 +134,24 @@ export default createMaterialTopTabNavigator(
   },
   {
     tabBarPosition: 'bottom',
+    animationEnabled: false,
+    swipeEnabled: false,
+    lazy: true,
```
Comment (dereckquock) [inline navigation/MainTabNavigator.js:139]: lazy load views!

File: navigation/MainTabNavigator.js [hunk 2]
```diff
@@ -141,16 +134,24 @@ export default createMaterialTopTabNavigator(
...
     optimizationsEnabled: true,
     tabBarOptions: {
       activeTintColor: Colors.tabIconSelected,
       inactiveTintColor: Colors.tabIconDefault,
       showIcon: true,
-      showLabel: false,
+      showLabel: true,
```
Comment (dereckquock) [inline navigation/MainTabNavigator.js:145]: show the labels 👍

File: screens/Groups.js [hunk 3]
```diff
@@ -1,19 +1,81 @@
...
+      { uuid: 'loading5' },
+      { uuid: 'loading6' },
+      { uuid: 'loading7' },
+      { uuid: 'loading8' },
+    ],
+    categories: [],
+  };
+
+  componentDidMount = async () => {
+    const [groups, categories] = await Promise.all([
```
Comment (dereckquock) [inline screens/Groups.js:73]: fetch the groups and categories

File: screens/Groups.js [hunk 4]
```diff
@@ -22,127 +84,129 @@ export default class GroupsScreen extends Component<Props> {
...
-            />
-          </View>
-          <Text style={styles.heading}>Echo Groups</Text>
-          <Text style={styles.text}>
-            One of the best ways for you to grow in your journey of faith,
-            connect in friendships with others, and serve the world around you.
-          </Text>
-        </View>
+      <View style={styles.mainContainer}>
+        <ImageBackground
```
Comment (dereckquock) [inline screens/Groups.js:88]: set a fullscreen image background

File: screens/Groups.js [hunk 5]
```diff
@@ -22,127 +84,129 @@ export default class GroupsScreen extends Component<Props> {
...
+          source={require('../assets/images/fall_leaves_bg.png')}
+          style={styles.backgroundImage}
+        />
+        <FlatList
```
Comment (dereckquock) [inline screens/Groups.js:92]: render groups in a flatlist

File: screens/Groups.js [hunk 6]
```diff
@@ -22,127 +84,129 @@ export default class GroupsScreen extends Component<Props> {
...
-            <Text style={styles.seasonLabel}>Summer:</Text>
-            <Text style={styles.seasonText}>Jun – Jul (sign-ups in May)</Text>
-          </View>
-          <View style={styles.season}>
-            <Text style={styles.seasonLabel}>Fall:</Text>
-            <Text style={styles.seasonText}>Sep – Dec (sign-ups in Aug)</Text>
-          </View>
-        </View>
-      </ScrollView>
+            groupTitleParts.shift();
```
Comment (dereckquock) [inline screens/Groups.js:111]: remove the group code thing from the group name title

we should probably do this on web too

File: screens/Groups.js [hunk 7]
```diff
@@ -22,127 +84,129 @@ export default class GroupsScreen extends Component<Props> {
...
+
+            return (
+              <View style={styles.card}>
+                {uuid.includes('loading') ? (
+                  <View style={styles.group}>
+                    <Loader />
+                  </View>
+                ) : (
+                  <View style={styles.group}>
+                    <Text
```
Comment (dereckquock) [inline screens/Groups.js:121]: the title should have a max number of lines of `2` and its text will adjust down to fit within those constraints

File: screens/Groups.js [hunk 8]
```diff
@@ -22,127 +84,129 @@ export default class GroupsScreen extends Component<Props> {
...
+                      <Text style={styles.detail}>
+                        {getMeetingTime(meetingTime)}
+                      </Text>
+                    </View>
+                    <View style={styles.details}>
+                      <Text style={[styles.detail, styles.bold]}>
+                        {getCampusCode(campus)}
+                      </Text>
+                    </View>
+                    <Text numberOfLines={6} style={styles.description}>
```
Comment (dereckquock) [inline screens/Groups.js:144]: `numberOfLines` truncates the description

File: App.js [hunk 9]
```diff
@@ -20,16 +19,19 @@ export default class App extends React.Component {
       );
     }
     return (
-      <SafeAreaView style={styles.container}>
```
Comment (Phillip3) [inline App.js:23]: Why did you move away from SafeAreaView? <- I like just using View (I think it's simpler), but SafeAreaView does seem to have some nice properties.

File: App.js [hunk 10]
```diff
@@ -54,6 +56,5 @@ export default class App extends React.Component {
 const styles = StyleSheet.create({
   container: {
     flex: 1,
-    backgroundColor: Colors.darkestGray,
```
Comment (Phillip3) [inline App.js:57]: I might rename the color situation here, but this seems legit for now :)

File: __tests__/__snapshots__/App-test.js.snap [hunk 11]
```diff
@@ -3,10 +3,9 @@
 exports[`App snapshot renders the loading screen 1`] = `null`;
```
Comment (Phillip3) [inline __tests__/__snapshots__/App-test.js.snap:3]: I have no idea what is going on here. This is all foreign to me, so let's sit down together sometime and look at this.

File: data/groups.js [hunk 12]
```diff
@@ -0,0 +1,179 @@
+// @flow
+
+import axios from 'axios';
+
+const env = process.env.NODE_ENV;
+const isDev = env !== 'production';
+
+// get the service endpoint based on the environment
+const baseURL = isDev
```
Comment (Phillip3) [inline data/groups.js:9]: Ternary operators are my love language.

File: data/groups.js [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+
+export async function getOpenGroups(): Object {
+  const today = new Date();
+  const year = today.getFullYear();
+  const startMonth = 5;
+  const startDay = 1;
+  const endMonth = 8;
+  const endDay = 19;
+
+  const {
```
Comment (Phillip3) [inline data/groups.js:22]: This syntax is funky af. What is happening here? Also, is there a more readable way to do this? Because this is kind of a mess--at least for someone who doesn't look at JS every day.

File: data/groups.js [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+    //   type: errorType,
+    //   stack: stackTrace,
+    // });
+    throw Error(errorMessage);
+  }
+
+  return groups.map(group => {
+    const {
+      campus = '',
+      categories = { CustomeCategories: [] },
```
Comment (Phillip3) [inline data/groups.js:64]: Custome... how fancy.

File: data/groups.js [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+      ...group,
+      campus: campus && campus.toUpperCase(),
+      attributes: categories.CustomeCategories,
+      hasChildcare: hasChildcare === 'true',
+    };
+  });
+}
+
+export async function getCategories(): Promise<Array<any>> {
+  const {
```
Comment (Phillip3) [inline data/groups.js:78]: This, being the same as 
https://github.com/Phillip3/the-resonance-project/blob/487fd9462c191452690a590909bc8ae24efa33e1/data/groups.js#L22-L39

but just having different inputs to axios.get might be worthwhile to combine.

File: navigation/MainTabNavigator.js (hunk 1 above)
Comment (Phillip3) [reply navigation/MainTabNavigator.js:139]: Shouldn't this be `createMaterialBottomTabNavigator`? 
https://github.com/Phillip3/the-resonance-project/blob/487fd9462c191452690a590909bc8ae24efa33e1/navigation/MainTabNavigator.js#L127

I think we don't want to do lazy loading. We should load all of the tab's initial content before the app opens. Waiting on a loading indicator when you hit each tab is annoying.

**Review by Phillip3 — CHANGES_REQUESTED** [review]
Mostly questions. A few things that should probably be changed. I didn't get to the groups part.

File: navigation/MainTabNavigator.js (hunk 1 above)
Comment (dereckquock) [reply navigation/MainTabNavigator.js:139]: it seems to work fine right now, even with the hundreds of groups loading.

lazy loading is good because the user's device can focus on the the current tab and when they open the next tab, then we'll load whatever's needed

File: App.js (hunk 9 above)
Comment (dereckquock) [reply App.js:23]: yeah, so `SafeAreaView` basically allows for the notch 😅 
but with the tab nav with the headers, we don't need it for now

File: App.js (hunk 10 above)
Comment (dereckquock) [reply App.js:57]: yeah, we can rename them, but i just made the names the actual colors lol 🤷‍♂

File: __tests__/__snapshots__/App-test.js.snap (hunk 11 above)
Comment (dereckquock) [reply __tests__/__snapshots__/App-test.js.snap:3]: yeah, so this snapshot is basically used for testing 🎉 
when running tests, it diffs the snapshot of the UI

https://jestjs.io/docs/en/snapshot-testing

File: data/groups.js (hunk 12 above)
Comment (dereckquock) [reply data/groups.js:9]: haha 😂

File: data/groups.js (hunk 13 above)
Comment (dereckquock) [reply data/groups.js:22]: haha it's readable for me 😂 
it's destructuring and it's a nice way to get stuff out of an object

File: data/groups.js (hunk 14 above)
Comment (dereckquock) [reply data/groups.js:64]: lol it was Axel

File: data/groups.js (hunk 15 above)
Comment (dereckquock) [reply data/groups.js:78]: ah, you mean the response? yeah, i guess we can pull that out into a function or something? 🤔

Comment (dereckquock) [reply data/groups.js:78]: meh...it's not that bad, so the overhead of pulling it out might be too much

File: navigation/MainTabNavigator.js (hunk 1 above)
Comment (one-tonne-mee) [reply navigation/MainTabNavigator.js:139]: i think we're refering to lazy-load for the visual after initial

**Review by Phillip3 — APPROVED** [review]
Woo!

**Review by one-tonne-mee — APPROVED** [review]
LGTM!
